src/metabase/models/model_index.clj

```diff
@@ -0,0 +1,150 @@
+(ns metabase.models.model-index
+  (:require
+   [cheshire.core :as json]
+   [clojure.string :as str]
+   [metabase.db.connection :as mdb.connection]
+   [metabase.mbql.normalize :as mbql.normalize]
+   [metabase.models.card :refer [Card]]
+   [metabase.models.interface :as mi]
+   [metabase.query-processor :as qp]
+   [metabase.sync.schedules :as sync.schedules]
+   [metabase.util.cron :as u.cron]
+   [metabase.util.i18n :refer [trs]]
+   [metabase.util.log :as log]
+   [toucan.models :as models]
+   [toucan2.core :as t2]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;; model lifecycle ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(models/defmodel ModelIndex :model_index)
+(models/defmodel ModelIndexValue :model_index_value)
+
+
+;; switch to the one in mi after #29513 lands
+(t2/define-before-insert ::created-at-timestamp
+  [instance]
+  #_{:clj-kondo/ignore [:private-call]}
+  (#'mi/add-created-at-timestamp instance))
+
+(derive ModelIndex ::created-at-timestamp)
+
+(def normalize-field-ref
+  "Normalize the field ref. Ensure it's well-formed mbql, not just json."
+  (comp #'mbql.normalize/canonicalize-mbql-clauses
+        #'mbql.normalize/normalize-tokens))
+
+(t2/deftransforms ModelIndex
```

Comment:
wdyt if we move this transformation to `model.interface` like this 

https://github.com/metabase/metabase/blob/9def91c91ff6427b76279f1a39ccdf2d4c4a0da7/src/metabase/models/interface.clj#L383-L386

Probably be useful in the future too.